### DISCUSSION TITLE
What is the format of using C Runtime Simulation Flags?

### DISCUSSION THREAD

**Discussion by ap766** [OpenModelica/OpenModelica#11402] [Q&A] [answered]
https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html#simflag-override

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/7a1093fb-e769-408c-a86c-5343723c18ba)

here's the particular command that I wish to run...

so what is the correct format 
is it like this?
`omc -override "tank1.A=2,tank2.V=7,startTime=2,stopTime=5" TwoConnectedTanks.mo`
(however I get error as Error: Unknown option -override.
Execution failed!

or 

`TwoConnectedTanks.exe -override=startTime=0,stopTime=5,tank1.A=9`

**Comment by ap766** [comment]
The second one doesnt show any changes .Only generated the .mat file but with the default values.I need it with the overridden values

**Comment by AnHeuermann** [comment, marked as answer]
You can supply simulation flags to the simulation executable only. The compiler `omc` only accepts [compiler flags](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html).

----------------------

So if you build model `Modelica.Fluid.Examples.AST_BatchPlant.Test.TwoTanks` using the following mos script

```modelica
loadModel(Modelica); getErrorString
buildModel(Modelica.Fluid.Examples.AST_BatchPlant.Test.TwoTanks); getErrorString();
```

you'll generate a simulation executable (`Modelica.Fluid.Examples.AST_BatchPlant.Test.TwoTanks.exe` on Windows or `Modelica.Fluid.Examples.AST_BatchPlant.Test.TwoTanks` on Linux).

Now you can call the executable with your desired flags:

```bash
Modelica.Fluid.Examples.AST_BatchPlant.Test.TwoTanks.exe -override=startTime=0,stopTime=100,stepSize=0.2,tolerance=1e-06
```

OMEdit will do this in a similar fassion and you can see the the call in the `Output` tab of the message browser.

**Reply by ap766** [reply]
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/74cc2171-004b-4c92-9eac-c66f5e166678)
so when i do it(2nd one) i get this.

The XML file shows no changes tho
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/cf0f9152-bc8e-4c15-acfe-d49c8930a7f9)

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/4108dcb3-e54e-4c7c-a088-7b70edc3a67e)
 and so this not reflected in the variables browser when I open the .mat file in openmodelica .

**Reply by AnHeuermann** [reply]
My answer explained how to use the simulation flags in general.
The `override` flag won't change the XML since it is not supposed to do that. It will override the values provided by the <model>_init.xm  file so the simulation will use the new (start, parameter,...) values instead. The updated values will be stored in the result file, but OMEdit isn't displaying those values in the `Value` section of the variables browser. When plotting the changed variable should show up, though.

So when running
```bash
./TwoTanks -override=tank1.level_start=2
```

and re-opening the result file in OMEdit I can see that the override was successful and the value of `tank1.level_start` was changed from `3` to `2`:
 
![grafik](https://github.com/OpenModelica/OpenModelica/assets/38031952/7ddbcbb3-019d-4684-a177-3a0b118dce88)


To my knowledge the simulation executable will never change the init_xml file, so there is no simulation flag to do that.

**Reply by ap766** [reply]
Ok,thank you so much!

So I suppose since the init_xml cant be changed , then `Value` would be showing the default only.

**Reply by ap766** [reply]
I suppose i could try changing the value of the XML file using ompython?So it reflects the value we had overridden?

**Reply by arun3688** [reply]
@ap766 If you are using `OMPython`  you don't need to change the xml file, you can simply use the API, 

`setParameters() and getParameters()` and follow the instructions in the user guide https://openmodelica.org/doc/OpenModelicaUsersGuide/1.21/ompython.html#enhanced-ompython-features

**Reply by ap766** [reply]
Oh okay, thanks a lot.

Yes I will go through it, just wanted to know if it would be possible to change the value in the value column in variables browser through ompython?

**Reply by ap766** [reply]
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/311f53e4-bc90-405e-838c-b45e1803c799)
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/2d671308-850a-46b8-957d-795cd063dbee)

 Also some problem here
